plugins/org.yakindu.sct.generator.cpp/src/org/yakindu/sct/generator/cpp/StatemachineHeader.xtend

```diff
@@ -170,6 +176,22 @@ class StatemachineHeader extends org.yakindu.sct.generator.c.files.StatemachineH
 		'''
 	}
 	
+	def createTracingScope(ExecutionFlow it) {
+		'''
+		class «tracingModule»
+		{
+			public:
+			virtual ~«tracingModule»() = 0;
+			
+			«IF entry.tracingEnterState»virtual void «enterStateTracingFctID»(«module»::«statesEnumType» state) = 0;«ENDIF»
+			
+			«IF entry.tracingExitState»virtual void «exitStateTracingFctID»(«module»::«statesEnumType» state) = 0;«ENDIF»
```

Comment:
Maybe an empty implementation that does nothing instead of pure virtual would be a good idea for all functions, this would allow the user to just provide a new instance of this class before he implements his own tracing class.